File: Data Structures & Algorithms/jump-game/submission-8.py
class Solution:
    def helper(self, nums, i):
        if i >= (len(nums) - 1):
            return True
        if nums[i] == 0:
            return False
        return True in [self.helper(nums, i + n) for n in range(1, nums[i] + 1)]

    def canJump(self, nums: List[int]) -> bool:
        return self.helper(nums, 0)

File: Data Structures & Algorithms/two-integer-sum/submission-8.cs
public class Solution {
    public int[] TwoSum(int[] nums, int target) {
        Dictionary<int, int> seen = new Dictionary<int, int>();
        int i = 0;
        foreach (int num in nums) {
            if (seen.ContainsKey(target - num)) {
                return [seen[target - num], i];
            }
            seen[num] = i;
            i++;
        }
        return [-1, -1];
    }
}
